008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,1014 @@
+    \item Sign the treaty if its overall benefit is positive.
+\end{enumerate}
+
+The amount of food that the “collectivist” and “selfish” agents would need to consume in order to maximise utility varies depending on the current health level. The peak of its total utility function thus needs to be able to vary as well. We account for this by introducing a scaling factor $a$ as
+
+\begin{equation}
+    U(x)=g(ax)^\frac{1}{r}-cax
+\end{equation}
+
+where
+
+\begin{equation}
+    a=\frac{1}{z}\left(\frac{cr}{g}\right)^\frac{r}{1-r}
+\end{equation}
+
+introducing the desired food intake $z$ as the maximum of the function without changing the general properties of the function, leading to:
+
+\begin{equation}
+U\left(x\right)=g\left(\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}x\right)^{\frac{1}{r}}-cx\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}    
+\end{equation}
+\subsection{Behaving According to Treaties}
+
+Treaties are fundamentally built on the assumption that all agents behave truthfully. Introducing dishonesty to treaties would sap them of all their power, as there are too many possible falsities but no sufficient way of recognising them and no central authority to punish them. Hence, we do not consider deception. Since this means that an agent cannot act against any active treaties, the amount of food taken has to be chosen with care.
+
+We implemented a function to return the valid range of possible food intakes given a list of active treaties and accepted request messages, and use this both to check if a newly proposed treaty is compatible with previously accepted treaties and to \textit{legally} take food. We do this as in Algorithm \ref{code:treatyRange}.
+% \Cref does not seem to work with algorithm components
+
+ \begin{algorithm}
+        \SetKwInOut{Input}{Input}
+        \SetKwInOut{Output}{Output}
+        \Input{List of active treaties \textbf{$T$}, RequestLeaveFood message \textbf{$M$}}
+        \Output{Maximum amount of food allowed: \textbf{max}}
+        \BlankLine
+        max $\gets$ food received \
+        \BlankLine
+        \ForEach{$t \in T$}
+        {
+        \uIf{\textup{agent state} $\in$ $t$ \textup{condition}}{
+            takeAmount $\gets$ convert $t$ request value
+            
+            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
+                max $\gets$ takeAmount
+            }
+
+        }
+        \BlankLine
+        \uIf{\textup{neighbour requested to leave food}}{
+            takeAmount $\gets$ convert $t$ request value
+            
+            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
+                max $\gets$ takeAmount
+            }
+        }            
+        }
+        \caption{Function to ensure that an agent never breaks any accepted treaties.}
+        \label{code:treatyRange}
+\end{algorithm}
+
+An agent then consumes the amount less than or equal to the resulting upper-bound \textit{max} that maximises their utility.
+
+
```

Comment:
\subsection{Short and Long Term Trade-Off}

As mentioned in \Cref{sec:forecasting}, the futures an agent should expect in the short (on the current level) and long term (throughout the tower) differ: the longer the duration of a treaty, the more the long-term benefit should matter. By using the short and long term memory of food received each day, an agent can separately estimate the resources it can expect to gain on the current floor and how much it can expect to gain (on average) across all future reassignment periods.

By calculating the length of the average reassignment period, the agents can estimate the number of days remaining on the current floor. The agents subsequently use the proportion of estimated days before the next reshuffling period to the duration of the treaty in order to weigh how much an agent should focus on the short term. Let $b_\mathit{short}$ and $b_\mathit{long}$ be the estimated long and short term benefit of a treaty, respectively. Also, let the estimated days remaining on the current level be given by $d_\mathit{current}$ and the duration of a treaty by $d_\mathit{treaty}$. The total benefit, $b_\mathit{tot}$ is given by

\begin{equation}
b_\mathit{tot} = d\times b_\mathit{short} + (1-d)\times b_\mathit{long}    
\label{eq:shortlongweight}
\end{equation}

where:

\begin{equation}
    d = \frac{d_\mathit{current}}{d_\mathit{treaty}}
\end{equation}

We supply the following example to illustrate this behaviour:

\begin{itemize}
    \item $d_\mathit{current} = 10$, $d_\mathit{treaty} = 8$:  $d=100\%$ (treaty expires before leaving the current floor)
    \item $d_\mathit{current} = 10$, $d_\mathit{treaty} = 20$: $d=50\%$
    \item $d_\mathit{current} = 10$, $d_\mathit{treaty} = 100$: $d = 10\%$
\end{itemize}


\subsubsection{Survival Instinct}

With a constant supply of food, the above weighting between the long and short term benefits is sensible, however if an agent is struggling to stay alive, it must redefine the way it weights these variables. Consequently, agents ignore the expected long-term utility altogether when their health is on a critical level.

\subsection{Implementation}

How agents rate treaties closely follows the theoretical steps from above. Each day, the agent stores the amount of food received in two arrays, one for both the short and long term, with the short-term array reset after each reshuffling period. When a new treaty is proposed, the agent performs the following steps to conclude whether to sign this new treaty or not:

\begin{enumerate}
    \item Checks if the trust in the \texttt{proposer} agent is high enough
    \item Checks if the treaty is compatible with treaties that the agent signed already
    \item Calculates the expected short- and long-term utility according to \Cref{eq:expectedUtility} using the memory of food received in the past. The utility function used depends on its current social motive.
    \item Amplifies the utility if it is negative in accordance with loss-aversion.
    \item Estimates the amount of food it can take under the treaty (i.e., the difference between the average amount of food it received in the past and the amount of food it needs to leave in order to comply with the treaty) and calculates the utility of taking that amount of food.
    \item Computes the estimated short- and long-term benefits of signing the treaty as $U(\text{sign}) - U(\text{don’t sign})$.
    \item Chooses to focus on the long-term or short-term benefit according to \Cref{eq:shortlongweight}
    \item Signs the treaty if its overall benefit is positive
\end{enumerate}

The amount of food that the collectivist and selfish agents would need to consume in order to maximise utility varies depending on the current health level. The peak of its total utility function thus needs to be able to vary, too. We account for this by introducing a scaling factor $a$ as:

\begin{equation}
    U(x)=g(ax)^\frac{1}{r}-cax
\end{equation}

where:

\begin{equation}
    a=\frac{1}{z}\left(\frac{cr}{g}\right)^\frac{r}{1-r}
\end{equation}

introducing the desired food intake $z$ as the maximum of the function without changing the general properties of the function, leading to:

\begin{equation}
U\left(x\right)=g\left(\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}x\right)^{\frac{1}{r}}-cx\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}    
\end{equation}
\subsection{Behaving According to Treaties}

Treaties are fundamentally built on the assumption that all agents behave truthfully. Introducing dishonesty to treaties would strap them of their power, as there are two many possible falsities but no sufficient way of recognising and no central authority to punish them. Hence, we do not consider deception: a decision shared by all agent teams. Since this means that an agent cannot act against any active treaties, the amount of food taken has to be chosen with care.

We implement a function to return the valid range of possible food intakes given a list of active treaties and accepted request messages and use this both to check if a newly proposed treaty is compatible with previously accepted treaties and to \textit{legally} take food. We do this as in Algorithm \ref{code:treatyRange}.
% \Cref does not seem to work with algorithm components

 \begin{algorithm}
        \SetKwInOut{Input}{Input}
        \SetKwInOut{Output}{Output}
        \Input{List of active treaties \textbf{$T$}, RequestLeaveFood message \textbf{$M$}}
        \Output{Maximum amount of food allowed: \textbf{max}}
        \BlankLine
        max $\gets$ food received \
        \BlankLine
        \ForEach{$t \in T$}
        {
        \uIf{\textup{agent state} $\in$ $t$ \textup{condition}}{
            takeAmount $\gets$ convert $t$ request value
            
            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
                max $\gets$ takeAmount
            }

        }
        \BlankLine
        \uIf{\textup{neighbour requested to leave food}}{
            takeAmount $\gets$ convert $t$ request value
            
            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
                max $\gets$ takeAmount
            }
        }            
        }
        \caption{Function to ensure that an agent never breaks any accepted treaties.}
        \label{code:treatyRange}
\end{algorithm}

An agent then consumes the quantity smaller or equal to the resulting upper-bound \texttt{max} that maximises its utility.
```suggestion
\subsection{Short and Long Term Trade-Off}

As mentioned in \Cref{sec:forecasting}, the futures the agent should expect in the short (on the current level) and long term (throughout the tower) differ. The longer the duration of a treaty, the more the long-term benefit should matter. By using the short and long term memory of food received each day, the agent can separately estimate the resources it can expect to gain on the current floor and how much it can expect to gain (on average) across all future reassignment periods.

By calculating the length of the average reassignment period, the agents can estimate the number of days remaining on the current floor. We use the proportion of estimated days before the next reshuffling period to the duration of the treaty in order to weigh how much an agent should focus on the short term. Let $b_\text{short}$ and $b_\text{long}$ be the estimated long and short term benefit of a treaty, respectively. Also, let the estimated days remaining on the current level be given by $d_\text{current}$ and the duration of a treaty by $d_\text{treaty}$. The total benefit $b_\text{tot}$ is given by

\begin{equation}
b_\text{tot} = d\times b_\text{short} + (1-d)\times b_\text{long}    
\label{eq:shortlongweight}
\end{equation}

where $d = \frac{d_\text{current}}{d_\text{treaty}}$.

The following examples show how this plays out.

\begin{itemize}
    \item $d_\text{current} = 10$, $d_\text{treaty} = 8\Rightarrow d=100\%$ (treaty expires before leaving the current floor)
    \item $d_\text{current} = 10$, $d_\text{treaty} = 20\Rightarrow d=50\%$
    \item $d_\text{current} = 10$, $d_\text{treaty} = 100\Rightarrow d = 10\%$
\end{itemize}


\subsubsection{Survival Instinct}

When one is doing well, the above weighting between the long and short term benefits is sensible. However, if one is struggling to stay alive, survival instincts kick in. Consequently, the agents ignore the expected long-term utility all together when their health is on a critical level.

\subsection{Implementation}

How agents rate treaties closely follows the theoretical steps from above. Every day, the agent stores the amount of food received in two arrays, one for the short and one for the long term. The short-term array is reset after every reshuffling period. When a new treaty is proposed, the agent performs the following steps to conclude whether to sign this new treaty or not:
\begin{enumerate}
    \item Check if the trust in the proposer agent is high enough.
    \item Check if the treaty is compatible with treaties that the agent signed already.
    \item Calculate the expected short- and long-term utility using the memory of food received in the past according to \Cref{eq:expectedUtility}. The utility function used depends on their current social motive.
    \item Amplify the utility if it is negative in accordance with loss-aversion.
    \item Estimate the amount of food it can take under the treaty, i.e., the difference between the average amount of food it received in the past and the amount of food it needs to leave in order to comply with the treaty, and calculate the utility of taking that amount of food.
    \item Compute the estimated short-term and long-term benefit of signing the treaty as $U(\text{sign}) - U(\text{don’t sign})$.
    \item Choose whether to focus on the long-term or short-term benefit according to \Cref{eq:shortlongweight}.
    \item Sign the treaty if its overall benefit is positive.
\end{enumerate}

The amount of food that the “collectivist” and “selfish” agents would need to consume in order to maximise utility varies depending on the current health level. The peak of its total utility function thus needs to be able to vary as well. We account for this by introducing a scaling factor $a$ as

\begin{equation}
    U(x)=g(ax)^\frac{1}{r}-cax
\end{equation}

where

\begin{equation}
    a=\frac{1}{z}\left(\frac{cr}{g}\right)^\frac{r}{1-r}
\end{equation}

introducing the desired food intake $z$ as the maximum of the function without changing the general properties of the function, leading to:

\begin{equation}
U\left(x\right)=g\left(\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}x\right)^{\frac{1}{r}}-cx\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}    
\end{equation}
\subsection{Behaving According to Treaties}

Treaties are fundamentally built on the assumption that all agents behave truthfully. Introducing dishonesty to treaties would sap them of all their power, as there are too many possible falsities but no sufficient way of recognising them and no central authority to punish them. Hence, we do not consider deception. Since this means that an agent cannot act against any active treaties, the amount of food taken has to be chosen with care.

We implemented a function to return the valid range of possible food intakes given a list of active treaties and accepted request messages, and use this both to check if a newly proposed treaty is compatible with previously accepted treaties and to \textit{legally} take food. We do this as in Algorithm \ref{code:treatyRange}.
% \Cref does not seem to work with algorithm components

 \begin{algorithm}
        \SetKwInOut{Input}{Input}
        \SetKwInOut{Output}{Output}
        \Input{List of active treaties \textbf{$T$}, RequestLeaveFood message \textbf{$M$}}
        \Output{Maximum amount of food allowed: \textbf{max}}
        \BlankLine
        max $\gets$ food received \
        \BlankLine
        \ForEach{$t \in T$}
        {
        \uIf{\textup{agent state} $\in$ $t$ \textup{condition}}{
            takeAmount $\gets$ convert $t$ request value
            
            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
                max $\gets$ takeAmount
            }

        }
        \BlankLine
        \uIf{\textup{neighbour requested to leave food}}{
            takeAmount $\gets$ convert $t$ request value
            
            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
                max $\gets$ takeAmount
            }
        }            
        }
        \caption{Function to ensure that an agent never breaks any accepted treaties.}
        \label{code:treatyRange}
\end{algorithm}

An agent then consumes the amount less than or equal to the resulting upper-bound \textit{max} that maximises their utility.

```suggestion
\subsection{Short and Long Term Trade-Off}

As mentioned in \Cref{sec:forecasting}, the futures the agent should expect in the short (on the current level) and long term (throughout the tower) differ. The longer the duration of a treaty, the more the long-term benefit should matter. By using the short and long term memory of food received each day, the agent can separately estimate the resources it can expect to gain on the current floor and how much it can expect to gain (on average) across all future reassignment periods.

By calculating the length of the average reassignment period, the agents can estimate the number of days remaining on the current floor. We use the proportion of estimated days before the next reshuffling period to the duration of the treaty in order to weigh how much an agent should focus on the short term. Let $b_\text{short}$ and $b_\text{long}$ be the estimated long and short term benefit of a treaty, respectively. Also, let the estimated days remaining on the current level be given by $d_\text{current}$ and the duration of a treaty by $d_\text{treaty}$. The total benefit $b_\text{tot}$ is given by

\begin{equation}
b_\text{tot} = d\times b_\text{short} + (1-d)\times b_\text{long}    
\label{eq:shortlongweight}
\end{equation}

where $d = \frac{d_\text{current}}{d_\text{treaty}}$.

The following examples show how this plays out.

\begin{itemize}
    \item $d_\text{current} = 10$, $d_\text{treaty} = 8\Rightarrow d=100\%$ (treaty expires before leaving the current floor)
    \item $d_\text{current} = 10$, $d_\text{treaty} = 20\Rightarrow d=50\%$
    \item $d_\text{current} = 10$, $d_\text{treaty} = 100\Rightarrow d = 10\%$
\end{itemize}


\subsubsection{Survival Instinct}

When one is doing well, the above weighting between the long and short term benefits is sensible. However, if one is struggling to stay alive, survival instincts kick in. Consequently, the agents ignore the expected long-term utility all together when their health is on a critical level.

\subsection{Implementation}

How agents rate treaties closely follows the theoretical steps from above. Every day, the agent stores the amount of food received in two arrays, one for the short and one for the long term. The short-term array is reset after every reshuffling period. When a new treaty is proposed, the agent performs the following steps to conclude whether to sign this new treaty or not:
\begin{enumerate}
    \item Check if the trust in the proposer agent is high enough.
    \item Check if the treaty is compatible with treaties that the agent signed already.
    \item Calculate the expected short- and long-term utility using the memory of food received in the past according to \Cref{eq:expectedUtility}. The utility function used depends on their current social motive.
    \item Amplify the utility if it is negative in accordance with loss-aversion.
    \item Estimate the amount of food it can take under the treaty, i.e., the difference between the average amount of food it received in the past and the amount of food it needs to leave in order to comply with the treaty, and calculate the utility of taking that amount of food.
    \item Compute the estimated short-term and long-term benefit of signing the treaty as $U(\text{sign}) - U(\text{don’t sign})$.
    \item Choose whether to focus on the long-term or short-term benefit according to \Cref{eq:shortlongweight}.
    \item Sign the treaty if its overall benefit is positive.
\end{enumerate}

The amount of food that the “collectivist” and “selfish” agents would need to consume in order to maximise utility varies depending on the current health level. The peak of its total utility function thus needs to be able to vary as well. We account for this by introducing a scaling factor $a$ as

\begin{equation}
    U(x)=g(ax)^\frac{1}{r}-cax
\end{equation}

where

\begin{equation}
    a=\frac{1}{z}\left(\frac{cr}{g}\right)^\frac{r}{1-r}
\end{equation}

introducing the desired food intake $z$ as the maximum of the function without changing the general properties of the function, leading to:

\begin{equation}
U\left(x\right)=g\left(\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}x\right)^{\frac{1}{r}}-cx\frac{1}{z}\left(\frac{cr}{g}\right)^{\frac{r}{1-r}}    
\end{equation}
\subsection{Behaving According to Treaties}

Treaties are fundamentally built on the assumption that all agents behave truthfully. Introducing dishonesty to treaties would sap them of all their power, as there are too many possible falsities but no sufficient way of recognising them and no central authority to punish them. Hence, we do not consider deception. Since this means that an agent cannot act against any active treaties, the amount of food taken has to be chosen with care.

We implemented a function to return the valid range of possible food intakes given a list of active treaties and accepted request messages, and use this both to check if a newly proposed treaty is compatible with previously accepted treaties and to \textit{legally} take food. We do this as in Algorithm \ref{code:treatyRange}.
% \Cref does not seem to work with algorithm components

 \begin{algorithm}
        \SetKwInOut{Input}{Input}
        \SetKwInOut{Output}{Output}
        \Input{List of active treaties \textbf{$T$}, RequestLeaveFood message \textbf{$M$}}
        \Output{Maximum amount of food allowed: \textbf{max}}
        \BlankLine
        max $\gets$ food received \
        \BlankLine
        \ForEach{$t \in T$}
        {
        \uIf{\textup{agent state} $\in$ $t$ \textup{condition}}{
            takeAmount $\gets$ convert $t$ request value
            
            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
                max $\gets$ takeAmount
            }

        }
        \BlankLine
        \uIf{\textup{neighbour requested to leave food}}{
            takeAmount $\gets$ convert $t$ request value
            
            \uIf{\textup{takeAmount} $\leq$ \textup{max}}{
                max $\gets$ takeAmount
            }
        }            
        }
        \caption{Function to ensure that an agent never breaks any accepted treaties.}
        \label{code:treatyRange}
\end{algorithm}

An agent then consumes the amount less than or equal to the resulting upper-bound \textit{max} that maximises their utility.


```
```